src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccount.fs

```diff
@@ -0,0 +1,69 @@
+﻿module Deciders.BankAccount
+
+open System
+open BankAccountPrimitives
+
+type BankAccountOpened =
+    { BankAccountId: AccountId
+      AccountNumber: AccountNumber
+      ClientId: ClientId
+      CurrencyISOCode: CurrencyCode
+      CreatedAt: DateTimeOffset
+      Version: int64 }
+
+type DepositRecorded =
+    { BankAccountId: AccountId
+      Amount: decimal
+      CashierId: CashierId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type CashWithdrawnFromATM =
+    { BankAccountId: AccountId
+      Amount: decimal
+      ATMId: AtmId
+      RecordedAt: DateTimeOffset
+      Version: int64 }
+
+type BankAccountClosed =
+    { BankAccountId: AccountId
+      Reason: string
+      ClosedAt: DateTimeOffset
+      Version: int64 }
+
+type Event =
+    | BankAccountOpened of BankAccountOpened
+    | DepositRecorded of DepositRecorded
+    | CashWithdrawnFromATM of CashWithdrawnFromATM
+    | BankAccountClosed of BankAccountClosed
+
+type BankAccount =
+    | NotInitialised
+    | Open of
+        {| Id: AccountId
```

Comment:
I probably do this too, but stuff will be much nicer if you stop using an anyonymous record type here
(put them at the end of the DU type defintions with `and OpenState = { id: AccountId; balance: decimal; version: int64 }`)

Reply:
Do you know some resources around where to select which style? I tried to understand when to use them, but so far, no luck. I've stolen this syntax from your PR in my samples 😎

Reply:
I believe the Stylish F# book covers these sorts of tradeoffs and is a good book in general (have only read a chapter or two so not a full assessment though all of Kit Eason's stuff anywhere is absolutely fantastic).

In general, as in writing, forward references (i.e. `and` is something to use sparingly). There's definitely a value to knowing that you can scan a file top to bottom to find things. If you pull it out in the above, I believe it will. help readablity in that one will be able to glance and see there are 3 cases, of which 2 have values. If you put the body record type defs immediately below, with or without a blank line in between, most readers can instantly get it whereas an 8 line block with inline anonymous record types is harder.

Another thing I tend to do is define records on a single line. The tradeoff between e.g. 3 vertical lines (which makes it harder to scan the code overall) vs having to pick out a field when 3 are on one line is obviously subjective, but I use it as a forcing function to make myself pull out sub-records if the line is getting long. Jumping straight to multi-line format just increases the temptation to whack everything into one record (with flags instead of modelling cases etc leaking in too when you go down that road!)

I believe there is an fslang-suggestions issue about relaxing the anomaly of not being able to construct anonymous records without special syntax. If that was implemented, there would be no negatives. However for a case body type that's used lots of times giving it a proper definition and name of its own is probably worth it.

Of course there are negatives to giving something a name that's pretty pointless - i.e. my suggestion doesnt make the code easier to read. Once the logic in the decider gets more interesting, you'll often find yourself wanting to utter the name of that body type anyway, which forces the issue.

So, in general, I tend to start with anonymous records but if I have more than 2 places instantiating them, I'll give them a name.